### ISSUE TITLE
Cannot open gpac after updating to MacOS 15.1

### ISSUE THREAD

**Issue by bnsimha** [gpac/gpac#3020]
I downloaded the stable version of gpac **(v2.4)** and tried to open the application on my **macbook air m1** after updating to **MacOS 15.1**

I got the following error while opening it through dock
> The application “Dock” does not have permission to open “(null)”.

I got the following error while opening it through finder
> The application “Finder” does not have permission to open “(null)”.

I tried re-downloading and re-installing but it doesn't seem to work. I even tried installing the latest v2.5 and I faced the same issue.

Even when trying to open the app through the terminal with the command 
```
Open -a  gpac
```
 I got the error - 
>_LSOpenURLsWithCompletionHandler() failed for the application /Applications/GPAC.app with error -54.

**Comment by aureliendavid** [comment]
Hi, 

(I don't have an M1 to test so these are just suggestions) 

There are several possibilities here. 

First of all, the pre-built mac installers that we provide are built for intel processors and may not work on the new apple silicon.  (we are investigating ways to provide arm-based binaries but it's a bit of a mess). 

You can try either: 

- running gpac through rosetta: https://medium.com/@lorenzozar/how-to-install-rosetta-on-mac-53c74391d2f5
- building gpac from source: https://wiki.gpac.io/Build/build/GPAC-Build-Guide-for-OSX

for example this person https://github.com/gpac/gpac/issues/2816 seems to have it working, but I don't know if they rebuilt from source or ran a pre-built package in the end

That being said, your specific issue looks like a permission issue and may not be related to processor architecture. 

This thread https://talk.macpowerusers.com/t/cryptic-error-when-running-ios-apps-on-m1/28864 seems to have some relevant info, or maybe this https://discussions.apple.com/thread/250794075?sortBy=rank

It may just be that some new security parameter prevents launching the app after updating. (where you able to run gpac on a m1 chip before?)

Sorry I'm not able to be more precise.

**Comment by rbouqueau** [comment]
It took the installers, both from 2.4 and the latest weekly ([gpac-2.5-DEV-rev729-gfa783afb-master.pkg](https://download.tsi.telecom-paristech.fr/gpac/new_builds/macosx_64/gpac-2.5-DEV-rev729-gfa783afb-master.pkg)). They both starts.

Nonetheless the apps are useless when used this way.

@bnsimha could you try to type `/Applications/GPAC.app/Contents/MacOS/MP4Box` from a terminal and let me know what it displays?

**Comment by aureliendavid** [comment]
> It took the installers, both from 2.4 and the latest weekly ([gpac-2.5-DEV-rev729-gfa783afb-master.pkg](https://download.tsi.telecom-paristech.fr/gpac/new_builds/macosx_64/gpac-2.5-DEV-rev729-gfa783afb-master.pkg)). They both starts.

thanks for the test 

so they open with `open -a gpac` on a m1 15.1?

I managed to reproduce the original issue on a x64 vm (so it's not an arch issue). 

It all worked on 15.0, then I updated to 15.1 and I get the same errors as OP with finder or `open -a gpac`

however I'm able to launch MP4Box/gpac from the command line, even for video playback with the gui

so it's probably an issue with SIP ; i'd like to try disabling it but I need to reboot to recovery and i'm not sure how to do that on a vm (the method I had before doesn't seem to work on the new version)

**Comment by aureliendavid** [comment]
yeah it looks like an issue with the 15.1 update, maybe someone can test if changing SIP settings solves it

@bradh you should still be able to run `gpac` and `MP4Box` from the command line (can you confirm?)

as a workaround, running `gpac -gui` is equivalent to launching the app via Finder or via `open -a gpac`

**Comment by bradh** [comment]
Probably meant for @bnsimha

**Comment by bnsimha** [comment]
Hello @aureliendavid

 I was able to use gpac and mp4box both before. 

 I want to play a 360 degree video with DASH SRD tiling, but now that **mp4client** is deprecated, I tried the gpac with the filters :

```
gpac -gui
```

```
gpac -play
```

but with both of them I got the 'compositor filter not connected' and other errors.

This led me to use gpac through the application. The application played my video by recognising the MPD link and I was also able to change my FOV with mouse movement by appending #VR to the MPD link. It showed me the list of tiles and their corresponding quality representations as well. Other things like bandwidth, buffer size etc were also displayed well. 

This was the only option I had as my video tiles are encoded in **hvt1.1.6.L186.80** codec. 

Should I open a new issue in this regard?

**Comment by bnsimha** [comment]
@rbouqueau 

I ran `/Applications/GPAC.app/Contents/MacOS/MP4Box`

I got the following response :

>Not enough arguments - check usage with -h
MP4Box - GPAC version 2.4-rev0-g5d70253a-master
(c) 2000-2024 Telecom Paris distributed under LGPL v2.1+ - https://gpac.io
Please cite our work in your research:
	GPAC Filters: https://doi.org/10.1145/3339825.3394929
	GPAC: https://doi.org/10.1145/1291233.129145

**Comment by aureliendavid** [comment]
> Probably meant for @bnsimha

wrong autocomplete, my bad, sorry to have bothered you



> I want to play a 360 degree video with DASH SRD tiling, but now that **mp4client** is deprecated, I tried the gpac with the filters :
> 
> ```
> gpac -gui
> ```
> 
> ```
> gpac -play
> ```
> 
> but with both of them I got the 'compositor filter not connected' and other errors.

can you post your exact command lines and resulting messages

can you try with 

```
gpac -gui <your_file>#VR
```

and

```
gpac -mp4c <your_file>#VR
```

and a standardized test with

```
gpac  -gui http://download.tsi.telecom-paristech.fr/gpac/gpac_test_suite/resources/media/counter/counter_1280_720_I_25_tiled_1mb.hevc#VR -r -logs=all@info
```

**Comment by bnsimha** [comment]
I tried all the three commands you told me. Here is what I got :

For 
```
gpac  -gui http://download.tsi.telecom-paristech.fr/gpac/gpac_test_suite/resources/media/counter/counter_1280_720_I_25_tiled_1mb.hevc#VR -r -logs=all@info
```

I got : 

>Refreshing list of modules in directory /opt/homebrew/Cellar/gpac/2.4.0/lib/gpac
[Core] Added module gm_validator.dylib.
Loaded 1 modules.
[Core] Plugin sdl of type 1196838712 not found in cache, searching for it...
[Core] Plugin sdl not found in 1 modules.
[Compositor] Failed to load a video output module.
Filter compositor failed to setup: I/O Error
Failed to load filter "compositor:player=gui": I/O Error
session error: I/O Error
[13:28:02.557Z] GPAC Session Status: mem          0 kb CPU  0
Active filters: 0
Filters not connected:
compositor (idx=1)
Filter compositor destruction

And for my own MPD file : 

Using **-gui** : 

>[Compositor] Failed to load a video output module.
Filter compositor failed to setup: I/O Error
Failed to load filter "compositor:player=gui": I/O Error
session error: I/O Error
Filters not connected:
compositor (idx=1)

Using **-mp4c** : 

>[Compositor] Failed to load a video output module.
Filter compositor failed to setup: I/O Error
Failed to load filter "compositor:player=base:src=1Pomerelle_Mountain_Resort-360_Video_6x4_2000_dash.mpd": I/O Error
session error: I/O Error
Filters not connected:
compositor (idx=1)

**Comment by aureliendavid** [comment]
you seem to have a previous install of gpac through homebrew that may be conflicting with the one you installed from the pkg. 

Can you try the following: 

```
which gpac
rm -rf ~/.gpac
gpac -h modules
gpac  -gui http://download.tsi.telecom-paristech.fr/gpac/gpac_test_suite/resources/media/counter/counter_1280_720_I_25_tiled_1mb.hevc#VR -r -logs=all@info
```

**Comment by bnsimha** [comment]
Yes, I had installed gpac through Homebrew. That was what I was working with. Should I remove it?

**Comment by aureliendavid** [comment]
did you also install it from a pkg downloaded from gpac.io? 

I didn't think the homebrew formula for gpac installed an app in /Applications/GPAC.app

can you run my commands above anyway just to check?

**Comment by bnsimha** [comment]
Yes I also downloaded it from gpac.io

I got this when I ran the commands : 

>[core] Creating default credential key in /Users/bhargavasimha/.gpac/creds.key, use -cred=PATH/TO_FILE to overwrite
Refreshing all options registry, this may take some time ... done
Available modules:
gm_validator.dylib: implements CompositorExtension (validator) 
Refreshing list of modules in directory /opt/homebrew/Cellar/gpac/2.4.0/lib/gpac
[Core] Added module gm_validator.dylib.
Loaded 1 modules.
[Core] Plugin sdl of type 1196838712 not found in cache, searching for it...
[Core] Plugin sdl not found in 1 modules.
[Compositor] Failed to load a video output module.
Filter compositor failed to setup: I/O Error
Failed to load filter "compositor:player=gui": I/O Error
session error: I/O Error
[17:39:50.227Z] GPAC Session Status: mem          0 kb CPU  0
Active filters: 0
Filters not connected:
compositor (idx=1)
Filter compositor destruction

**Comment by aureliendavid** [comment]
(the result of the other commands would have been interesting too)

I think something is broken between both install, you should probably

- remove both versions of the gpac (homebrew and app package)
- delete your ~/.gpac directory (again)
- reinstall the latest version from https://download.tsi.telecom-paristech.fr/gpac/new_builds/gpac_latest_head_macos.pkg
- retest

**Comment by bnsimha** [comment]
After having removed the homebrew version and installing the latest version from the link `download.tsi.telecom-paristech.fr/gpac/new_builds/gpac_latest_head_macos.pkg`, When I type :

```
which gpac 
```

or 

```
which mp4box
```

I get `gpac not found` and `mp4box not found` respectively. But the app is visibly installed and inaccessible, regressing to the reason of this github issue.

Only With the homebrew installation was I able to access gpac through the command line (though it was on `/opt/homebrew/` as opposed to `/Applications/`)

**Comment by aureliendavid** [comment]
this is very strange and not at all the behavior i'm getting 

does the file /Applications/GPAC.app/Contents/MacOS/MP4Box exists after installing the downloaded package? can you run it?

**Comment by aureliendavid** [comment]
in any case, if you can't make the installers work, I would encourage you to rebuild gpac from source by following the guide at https://wiki.gpac.io/Build/build/GPAC-Build-Guide-for-OSX/#full-build

it's the method that has the best chances to work well on your system

**Comment by bnsimha** [comment]
I added the GPAC path `/Applications/GPAC.app/Contents/MacOS` to the PATH variable and its working now. I am able to access **mp4box** and **gpac** through command-line !!!!! 

But yeah opening the application through `Finder` or `Dock` is still not possible.

**Comment by bnsimha** [comment]
@aureliendavid Thank you for helping me out!!

**Comment by rbouqueau** [comment]
Is this issue still relevant for new installs?

**Comment by frozenspider** [comment]
Tried it with the latest build, and yes, the error I'm getting when installing GPAC through installer is exactly the same, including message when trying to open it via CLI.
I'm using MacBook Pro M1 with macOS 15.7.3 (24G419) installed.

Building from source didn't work either - `make` after `./configure --static-bin` complained about `_FT_Done_Face` and other freetype symbols being undefined (despite freetype being installed in the system), and for the full build, I failed to build `deps_unix` because CMake is of outdated version and `./libavutil/timer.h:42:13: fatal error: 'aarch64/timer.h' file not found`

**Comment by rbouqueau** [comment]
@frozenspider I'm on the same MacOS version as you. I see the issue. Reopening until I find some time to investigate. Thanks for insisting.

**Comment by rbouqueau** [comment]
@frozenspider I was talking about the installer. Can you open a separate issue about your build not executing? We've had users complaining about fonts not being installed (but obviously this doesn't happen on any maintainer's computer nor on our build inftrastructure - it would be too easy). Thanks.

**Comment by rbouqueau** [comment]
That's a codesign issue:
```
rbouqueau@didouce ~ % codesign --remove-signature /Applications/GPAC.app 
rbouqueau@didouce ~ % codesign --force --deep --sign - /Applications/GPAC.app
```

@aureliendavid Do we do anything specific to the dmg before uploading it?

FYI the initial error disappeared, but clicking on GPAC would still silently fail. The logo appears on the Dock, bounces for a few seconds, then it closes. I tried to open it from the Terminal with `open /Applications/GPAC.app` but it doesn't display anything (not even the bouncing icon on the Dock - which doesn't seem a reliable sign at all).
